generators/generator-base.js

```diff
@@ -2434,4 +2434,24 @@ module.exports = class extends PrivateBase {
         dest.fieldNameChoices = [];
         dest.relNameChoices = [];
     }
+
+    /**
+     * Get all the generator configuration from the .yo-rc.json file
+     * @param {Generator} generator the generator instance to use
+     * @param {boolean} force force getting direct from file
+     */
+    getAllJhipsterConfig(generator = this, force) {
+        let configuration = generator.config.getAll() || {};
+        if ((force || !configuration.baseName) && jhiCore.FileUtils.doesFileExist('.yo-rc.json')) {
+            configuration = JSON.parse(fs.readFileSync('.yo-rc.json', { encoding: 'utf-8' }))['generator-jhipster'];
+        }
+        if (!configuration.get || typeof configuration.get !== 'function') {
+            Object.assign(configuration, {
```

Comment:
You're trying my previous solution, I suggest you try my new one.
If you need the configuration in other generators without having to pass it around, I suggest you use the `this.createConfigFromNewConfFile` (may have to rename it).
You're simulating a Storage class, and it may work but isn't valid in some cases.

Reply:
I wonder if you still need to do this by the way... with the method being plugged in the top class.

Reply:
@MathieuAA yes indeed my solution is a bit hacky and frankly, i'm not very proud of this. I saw your solution only after while resolving conflicts on this PR yesterday, but i'm quite tight on time, so if you have any spare time will you be kind enough to update this? else i'll do it later when I have time to study yours and use it. Also please note that the idea here is to always load the config ourself without relying on yeoman, so that we get the JHipster config all the time and not the blueprints config(so we might have to hardcode the `generator-jhipster` key in your solution as well

Reply:
Agreed!